pygeos/predicates.py

```diff
@@ -637,8 +635,12 @@ def intersects(a, b, **kwargs):
 
 @multithreading_enabled
 def overlaps(a, b, **kwargs):
-    """Returns True if A and B intersect, but one does not completely contain
-    the other.
+    """Returns True if A and B spatially overlap.
+
+    A and B overlap if they have some but not all points in common, have the
+    same dimension, and the intersection of the interiors of the two geometries
+    has the same dimension as the geometries themselves.  That is, only polyons
+    can overlap other polygons and only lines can overlap other lines.
```

Comment:
And eg lines that only cross at a single point don't overlap? (that might be useful to add as well to illustrate the "same dimension" rule?)